src/planning/src/DijkstraPlanner.cpp

```diff
@@ -0,0 +1,436 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+// Navigation Strategy based on:
+// Brock, O. and Oussama K. (1999). High-Speed Navigation Using the Global Dynamic Window Approach. IEEE.
+// https://cs.stanford.edu/group/manips/publications/pdfs/Brock_1999_ICRA.pdf
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <chrono>
+#include <limits>
+#include <iostream>
+#include <algorithm>
+#include "planning/DijkstraPlanner.hpp"
+#include "planning/Navfn.hpp"
+#include "costmap/CostValues.h"
+
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "geometry_msgs/msg/point.hpp"
+#include "nav2_msgs/msg/costmap.hpp"
+#include "nav2_msgs/srv/get_costmap.hpp"
+
+using namespace std::chrono_literals;
+
+
+DijkstraPlanner::DijkstraPlanner(const std::string & name)
+: ComputePathToPoseTaskServer(name), global_frame_("/map"), allow_unknown_(true), default_tolerance_(1.0)
+{
+  RCLCPP_INFO(get_logger(), "DijkstraPlanner::DijkstraPlanner");
+
+  //TODO(orduno): Enable parameter server
+
+  costmap_client_ = this->create_client<nav2_msgs::srv::GetCostmap>("CostmapService");
+  waitForCostmapServer();
+
+  // TODO(orduno): Service for getting the costmap sometimes fails, check how the parent task can handle this.
+  //               might need to pull out the getCostmap() method out of the constructor and have an initialize() one.
+  try {
+    getCostmap(costmap_);
+  }
+  catch (...) {
+    RCLCPP_ERROR(this->get_logger(), "DijkstraPlanner::makePlan: failed to obtain costmap from server");
+    throw;
+  }
+
+  printCostmap(costmap_);
+
+  planner_ = std::make_shared<NavFn>(costmap_.info.width, costmap_.info.height);
+
+  // Plan publisher for visualization purposes
+  plan_publisher_ = this->create_publisher<nav2_msgs::msg::Path>("plan", 1);
```

Comment:
Did you intend to include the second parameter (qos_history_depth)? Or, is this a left-over from porting?

Reply:
left-over from porting